### DISCUSSION TITLE
Quasar 2.0 .vue file use boot file - function

### DISCUSSION THREAD

**Discussion by ecmap** [quasarframework/quasar#9806] [General - Components / Directives / etc] [answered]
I am confused to use boot file in .vue 

suppose I want to use moment

and 
1. install moment
2. quasar new boot moment
```
import { boot } from 'quasar/wrappers'
import moment from 'moment'

// "async" is optional;
// more info on params: https://v2.quasar.dev/quasar-cli/boot-files
export default boot(async ({ app }) => {
  app.config.globalProperties.$moment = moment
})

export { moment }
```
3. boot: ['moment'], in quasar.conf.js

4.in .vue file (tried moment, $moment) - result: XXXX is not defined
```
import { defineComponent,onMounted } from 'vue';

export default defineComponent({
  name: 'indexPage',
  setup () {
    onMounted(() => {
      console.log(moment)
    })
  }
})
```

how to fix it? 

This is my first time using quasar v2 and vue 3

(quasar v1 user)

**Comment by rstoenescu** [comment, marked as answer]
Just import "moment" where you want to use it directly in your .vue files.
The moment library does not require any setup so a boot file would have limited to no purpose.

If you want to supply the "moment" object in your app (without having to import it in every file that you'll be using it), then:
1. If using Composition API -- you need to use the provide/inject mechanism in Vue (`app.provide('moment', moment)` in your boot file), but in this case you will be required to import "inject" from "vue", so it defeats the purpose (which is the ease of use, not having to import something)
2. If using Options API -- the `app.config.globalProperties.$moment = moment` will make `this.$moment` available

PS: You might want to look at Quasar's own [date utils](https://quasar.dev/quasar-utils/date-utils)

**Reply by ecmap** [reply]
thanks
